libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -375,13 +385,19 @@ export function OutQueueManager(
           clearTimeout(xhrTimeout);
           if (xhr.status >= 200 && xhr.status < 300) {
             onPostSuccess(numberToSend);
-          } else {
-            if (!shouldRetryForStatusCode(xhr.status)) {
-              LOG.error(`Status ${xhr.status}, will not retry.`);
-              removeEventsFromQueue(numberToSend);
-            }
-            executingQueue = false;
+            return;
+          }
+
+          if (xhr.status === 0) {
```

Comment:
Why do we handle the status code 0 differently from other status codes? Shouldn't the `retryFailedRequests` also apply for 4xx and 5xx status codes (and override any custom retry code setting)? I would make it a part of `shouldRetryForStatusCode` function to check it – we do it similarly in the mobile trackers, [see here](https://github.com/snowplow/snowplow-ios-tracker/blob/master/Sources/Snowplow/Network/RequestResult.swift#L60-L63).